core/src/com/mygdx/game/client/ecs/system/UnitMovementSystem.java

```diff
@@ -25,11 +27,14 @@ public class UnitMovementSystem extends IteratingSystem {
   private final ComponentMapper<UnitMovement> unitMovementMapper = getFor(UnitMovement.class);
   private final ComponentMapper<Position> positionMapper = getFor(Position.class);
   private final ComponentMapper<Slot> slotMapper = getFor(Slot.class);
+  private final ComponentMapper<Score> scoreMapper = getFor(Score.class);
+  private final PlayerScore playerScore;
 
   @Inject
-  public UnitMovementSystem() {
+  public UnitMovementSystem(@NonNull PlayerScore playerScore) {
     super(Family.all(UnitMovement.class, Position.class).get());
 //    this.movementSyncer = movementSyncer;
+    this.playerScore = playerScore;
```

Comment:
Ogólnie nie miałem pomysłu gdize przechowywać ten wynik więc stworzylem taką klasę PlayerScore, znajduje się ona obok GameState, chciałem to wrzucić w GameState ale robiła się pętla dependencji, potem można zmienić, ale narazie moze być tutaj

Reply:
Wydaje mi się, że DI bardziej pomaga gdy faktycznie jakieś klasy są potrzebne, a nie gdy są injectowane "na zaś", albo dlatego, że konceptualnie pasuje, żeby były razem. Ale nie wiem jak to wyglądało.